### PR TITLE
Modification of 'Adding modules and plugins locally' topic

### PR DESCRIPTION
**Pull request by sayee-jadhav**
SUMMARY
Incorporates writing style changes per IBM style guide change and minimalism practices. This PR is the next step. There's more that can be improved here, but each iteration is a step forward.

ISSUE TYPE
Docs Pull Request

COMPONENT NAME
docs.ansible.com
modules
plugins

### CODE REVIEW

**Comment by sayee-jadhav** [comment]
Request review from @acozine and @samccann

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 1]
```diff
@@ -5,41 +5,47 @@
...
-To save a local module or plugin so Ansible can find and use it, drop the module or plugin in the correct "magic" directory. For local modules, use the name of the file as the module name: for example, if the module file is ``~/.ansible/plugins/modules/local_users.py``, use ``local_users`` as the module name.
+To save a local module or plugin such that Ansible can find and use it, add the module or plugin in the appropriate directory (the directories are specified in later parts of this topic). 
+
+.. contents::
+   :local:
 
 .. _modules_vs_plugins:
 
-Modules and plugins: what's the difference?
+What are modules and plugins?
```
Comment (Akasurde) [inline docs/docsite/rst/dev_guide/developing_locally.rst]: I would keep the original one.

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 2]
```diff
@@ -5,41 +5,47 @@
...
 ===========================================
-If you're looking to add local functionality to Ansible, you may be wondering whether you need a module or a plugin. Here's a quick overview of the differences:
+If you're looking to add local functionality to Ansible, you might wonder whether you need a module or a plugin. Here's a quick overview to help you decide between the two:
 
-* Modules are reusable, standalone scripts that can be used by the Ansible API, the :command:`ansible` command, or the :command:`ansible-playbook` command. Modules provide a defined interface, accepting arguments and returning information to Ansible by printing a JSON string to stdout before exiting. Modules execute on the target system (usually that means on a remote system) in separate processes.
+* Modules are reusable, standalone scripts that can be used by the Ansible API, the :command:`ansible` command, or the :command:`ansible-playbook` command. Modules provide a defined interface, accept arguments, and return information to Ansible by printing a JSON string to stdout before exiting. Modules execute on the target system (usually that means on a remote system) in separate processes.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]: This change altered the meaning slightly, but the sentence could definitely be clarified. How about:
```suggestion
* Modules are reusable, standalone scripts that can be used by the Ansible API, the :command:`ansible` command, or the :command:`ansible-playbook` command. Modules provide a defined interface. Each module accepts arguments and returns information to Ansible by printing a JSON string to stdout before exiting. Modules execute on the target system (usually that means on a remote system) in separate processes.
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 3]
```diff
@@ -5,41 +5,47 @@
 Adding modules and plugins locally
 **********************************
 
-.. contents::
-   :local:
+The easiest, quickest, and the most popular way to extend Ansible is to use a module or a plugin. You can create them or copy existing ones for local use. You can store a local module or plugin on your Ansible control node and share it with your team or organization. You can also share a local plugin or module by embedding it in a role and publishing it on Ansible Galaxy. If you are using roles on Ansible Galaxy, then you are already using local modules and plugins without realizing it.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]: The first sentence here, as changed, is not quite accurate. Modules and plugins are the building blocks of Ansible behavior. This page shows users how to get a **local** plugin or module to work.
```suggestion
The easiest, quickest, and the most popular way to extend Ansible is to use a local module or a plugin. You can create them or copy existing ones for local use. You can store a local module or plugin on your Ansible control node and share it with your team or organization. You can also share a local plugin or module by embedding it in a role and publishing it on Ansible Galaxy. If you are using roles on Ansible Galaxy, then you are already using local modules and plugins without realizing it.
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 4]
```diff
@@ -51,22 +57,19 @@ or
 
 .. note::
 
-	Currently, ``ansible-doc`` command can only parse Python modules for the module documentation. If you have module written in a different programming language other than Python, please write a documentation in Python file adjacent to module file.
-
-To use a local module only in certain playbooks:
+	Currently, the ``ansible-doc`` command can parse Python modules only for the module documentation. If you have a module written in a different programming language other than Python, please write the documentation in a Python file adjacent to the module file.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]: Nice catch here. This section needed revision. The placement of `only` in the revised version feels awkward, though. How about this:
 ```suggestion
	Currently, the ``ansible-doc`` command can parse module documentation only from modules written in Python. If you have a module written in a programming language other than Python, please write the documentation in a Python file adjacent to the module file.
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 5]
```diff
@@ -51,22 +57,19 @@ or
...
 
-* store it in a sub-directory called ``library`` in the directory that contains the playbook(s)
+The location of your module depends on whether you want to use it in a playbook or in a single role. Choose any of the following locations to add your module: 
 
-To use a local module only in a single role:
-
-* store it in a sub-directory called ``library`` within that role
+* In a certain playbook: Store the module in a subdirectory called ``library`` in the directory that contains the playbooks.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]:
```suggestion
* In a selected playbook or playbooks: Store the module in a subdirectory called ``library`` in the directory that contains those playbooks.
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 6]
```diff
@@ -80,24 +83,23 @@ Ansible loads plugins automatically too, loading each type of plugin separately
...
 
 * any directory added to the relevant ``ANSIBLE_plugin_type_PLUGINS`` environment variable (these variables, such as ``$ANSIBLE_INVENTORY_PLUGINS`` and ``$ANSIBLE_VARS_PLUGINS`` take colon-separated lists like ``$PATH``)
 * the directory named for the correct ``plugin_type`` within ``~/.ansible/plugins/`` - for example, ``~/.ansible/plugins/callback``
 * the directory named for the correct ``plugin_type`` within ``/usr/share/ansible/plugins/`` - for example, ``/usr/share/ansible/plugins/action``
 
-Once your plugin file is in one of these locations, Ansible will load it and you can use it in a any local module, task, playbook, or role. Alternatively, you can edit your ``ansible.cfg`` file to add directories that contain local plugins - see :ref:`ansible_configuration_settings` for details.
+After your plugin file is in one of these locations, Ansible loads it and you can use it in any local module, task, playbook, or role. Alternatively, you can edit your ``ansible.cfg`` file to add directories that contain local plugins. For information about adding directories of local plugins, see :ref:`ansible_configuration_settings` for details.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]:
```suggestion
After your plugin file is in one of these locations, Ansible loads it and you can use it in any local module, task, playbook, or role. Alternatively, you can edit your ``ansible.cfg`` file to add directories that contain local plugins. For details about adding directories of local plugins, see :ref:`ansible_configuration_settings`.
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 7]
```diff
@@ -51,22 +57,19 @@ or
 
 .. note::
 
-	Currently, ``ansible-doc`` command can only parse Python modules for the module documentation. If you have module written in a different programming language other than Python, please write a documentation in Python file adjacent to module file.
-
-To use a local module only in certain playbooks:
+	Currently, the ``ansible-doc`` command can parse Python modules only for the module documentation. If you have a module written in a different programming language other than Python, please write the documentation in a Python file adjacent to the module file.
 
-* store it in a sub-directory called ``library`` in the directory that contains the playbook(s)
+The location of your module depends on whether you want to use it in a playbook or in a single role. Choose any of the following locations to add your module:
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]: This list is about limiting the scope/context for a local module. 
```suggestion
You can limit the availability of your local module. If you want to use a local module only with selected playbooks or only with a single role, load it in one of the following locations:
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 8]
```diff
@@ -5,41 +5,47 @@
...
 .. _local_modules:
 
 Adding a module locally
 =======================
-Ansible automatically loads all executable files found in certain directories as modules, so you can create or add a local module in any of these locations:
+Ansible automatically loads all executable files found in certain directories as modules.
+
+For local modules, use the name of the file as the module name: for example, if the module file is ``~/.ansible/plugins/modules/local_users.py``, use ``local_users`` as the module name.
+
+To load your local modules automatically, add them in any of these locations:
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]:
```suggestion
To load your local modules automatically and make them available to all playbooks and roles, add them in any of these locations:
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 9]
```diff
@@ -80,24 +83,23 @@ Ansible loads plugins automatically too, loading each type of plugin separately
...
 * type ``ansible-doc -t <plugin_type> my_custom_lookup_plugin``. For example, ``ansible-doc -t lookup my_custom_lookup_plugin``. You should see the documentation for that plugin. This works for all plugin types except the ones marked with ``*`` in the list above  - see :ref:`ansible-doc` for more details.
 
-To use your local plugin only in certain playbooks:
-
-* store it in a sub-directory for the correct ``plugin_type`` (for example, ``callback_plugins`` or ``inventory_plugins``) in the directory that contains the playbook(s)
-
-To use your local plugin only in a single role:
-
-* store it in a sub-directory for the correct ``plugin_type`` (for example, ``cache_plugins`` or ``strategy_plugins``) within that role
+The location of your plugin depends on whether you want to use it in a playbook or in a single role.  Choose any of the following locations to add your plugin:
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]: Same comment here as above - this is about limiting the scope of a plugin.
```suggestion
You can limit the availability of your local plugin. If you want to use a local plugin only with selected playbooks or only with a single role, load it in one of the following locations:
```

File: docs/docsite/rst/dev_guide/developing_locally.rst [hunk 10]
```diff
@@ -80,24 +83,23 @@ Ansible loads plugins automatically too, loading each type of plugin separately
...
 
-When shipped as part of a role, the plugin will be available as soon as the role is called in the play.
+* In a certain playbook: Store the plugin in a subdirectory for the correct ``plugin_type`` (for example, ``callback_plugins`` or ``inventory_plugins``) in the directory that contains the playbooks.
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/developing_locally.rst]:
```suggestion
* In a selected playbook or playbooks: Store the plugin in a subdirectory for the correct ``plugin_type`` (for example, ``callback_plugins`` or ``inventory_plugins``) in the directory that contains the playbooks.
```

**Review by acozine — COMMENTED** [review]
Thanks @sayee-jadhav, this cleans up quite a few stylistic elements on this page. I've added some comments to make the changes even stronger.

File: docs/docsite/rst/dev_guide/developing_locally.rst (hunk 1 above)
Comment (sayee-jadhav) [reply docs/docsite/rst/dev_guide/developing_locally.rst]: Thank you for the review, @Akasurde. I updated the title in the commit instance 0880239.

**Review by sayee-jadhav — COMMENTED** [review]
Thank you for the review, @acozine. Your feedback makes sense and +1 for merging them. Request to merge this PR.

**Pull request merged**